src/sentry/ratelimits/redis.py

```diff
@@ -15,6 +17,27 @@ class RedisRateLimiter(RateLimiter):
     def __init__(self, **options):
         self.cluster, options = get_cluster_from_options("SENTRY_RATELIMITER_OPTIONS", options)
 
+    def _construct_redis_key(
+        self, key: str, project: Optional[Project] = None, window: Optional[int] = None
+    ) -> str:
+        """
+        Construct a rate limit key using the args given. Key will have a format of:
+        "rl:<key_hex>:[project?<project_id>:]<time_bucket>"
+        where the time bucket is calculated by integer dividing the current time by the window
+        """
+
+        if window is None:
```

Comment:
I'd make this 
```py
if window:
    ...
```
just to get extra safety incase someone accidentally passes in zero.

Reply:
any objections to `if window is None or window == 0`?